tests/test_unit/test_uncertainty_evaluators.py

```diff
@@ -1,159 +1,126 @@
 """Chemprop unit tests for chemprop/uncertainty/uncertainty_evaluator.py"""
-from unittest import TestCase
+import uuid
+
 import numpy as np
+import pytest
+
+from chemprop.uncertainty.uncertainty_evaluator import build_uncertainty_evaluator
+
+
+@pytest.fixture(params=["regression", "classification", "multiclass"])
+def dataset_type(request):
+    return request.param
+
+
+@pytest.fixture(params=["nll", "miscalibration_area", "ence", "spearman"])
+def regression_metric(request):
+    return request.param
+
+
+@pytest.fixture(params=["auc", "prc-auc", "accuracy", "f1", "mcc"])
+def classification_metric(request):
+    return request.param
+
+
+@pytest.fixture(params=["cross_entropy", "accuracy", "f1", "mcc"])
+def multiclass_metric(request):
+    return request.param
+
+
+@pytest.fixture
+def nll_regression_evaluator():
+    return build_uncertainty_evaluator("nll", None, "ensemble", "regression", "mse", None)
+
+
+@pytest.fixture
+def nll_classification_evaluator():
+    return build_uncertainty_evaluator(
+        "nll", None, "classification", "classification", "binary_cross_entropy", None
+    )
+
+
+@pytest.fixture
+def miscal_regression_evaluator():
+    return build_uncertainty_evaluator(
+        "miscalibration_area", None, "ensemble", "regression", "mse", None
+    )
+
+
+@pytest.fixture
+def spearman_evaluator():
+    return build_uncertainty_evaluator("spearman", None, "ensemble", "regression", "mse", None)
+
+
+def test_build_regression_metric(regression_metric):
+    assert build_uncertainty_evaluator(regression_metric, None, None, "regression", None, None)
+
+
+def test_build_classification_metric(classification_metric):
+    assert build_uncertainty_evaluator(
+        classification_metric, None, None, "classification", None, None
+    )
+
+
+def test_build_multiclass_metric(multiclass_metric):
+    assert build_uncertainty_evaluator(multiclass_metric, None, None, "multiclass", None, None)
+
+
+@pytest.mark.parametrize("metric", [str(uuid.uuid4()) for _ in range(3)])
+def test_build_unsupported_metrics(metric, dataset_type):
+    with pytest.raises(NotImplementedError):
+        build_uncertainty_evaluator(metric, None, None, "spectra", dataset_type, None)
```

Comment:
I don't understand what's going on in this block. Is "metric" just a string or is pytest doing something funny with it where it becomes the name of an attribute or something?

And is the uuid just putting gibberish that then becomes dataset_type (though dataset_type is in the loss_function spot).

This is all just to trigger L398 from the "spectra" entry, so why the gibberish?